### PR TITLE
Predicting Loan E-Signatures: Add Boosting Algorithms

### PR DESCRIPTION
**Pull request by c-grigsby**
Updated feature engineering & added XGBoost and scikit-learn gradient boosting algorithms for use case Predicting Loan E-Signatures. Hyperopt trials documented via MLflow. Implemented a workflow for final model inference and evaluation.

### CODE REVIEW

**Comment by c-grigsby** [comment]
Looks good to me :rocket:
